### PR TITLE
Automatically generate enum values and avoid hash lookup

### PR DESCRIPTION
**Pull request by geekbeast**

### CODE REVIEW

**Comment by makosblade** [comment]
Integrated into https://github.com/geekbeast/rhizome/pull/297

File: src/main/kotlin/com/openlattice/hazelcast/serializers/AbstractEnumSerializer.kt [hunk 1]
```diff
@@ -5,13 +5,13 @@ import com.google.common.cache.CacheLoader
...
+import org.apache.commons.lang.math.RandomUtils
 
 abstract class AbstractEnumSerializer<T : Enum<T>> : TestableSelfRegisteringStreamSerializer<Enum<T>> {
+    protected val enumArray = enumCache.getOrPut(clazz) { this.clazz.enumConstants as Array<Enum<*>> } as Array<Enum<T>>
 
     companion object {
-        private val enumCache: LoadingCache<Class<*>, Array<*>> = CacheBuilder.newBuilder().build( CacheLoader.from { key ->
-            key!!.enumConstants
-        })
+        private val enumCache: MutableMap<Class<*>, Array<Enum<*>>> = mutableMapOf()
```
Comment (dbt) [inline src/main/kotlin/com/openlattice/hazelcast/serializers/AbstractEnumSerializer.kt]: mutableMapOf() returns a LinkedHashMap, which isn't threadsafe. Either use concurrent hashmap or synchronized hashmap.

**Comment by makosblade** [comment]
Also I have moved this class to Rhizome already 😬

File: src/main/kotlin/com/openlattice/hazelcast/serializers/AbstractEnumSerializer.kt (hunk 1 above)
Comment (makosblade) [reply src/main/kotlin/com/openlattice/hazelcast/serializers/AbstractEnumSerializer.kt]: fixed here: https://github.com/geekbeast/rhizome/pull/291/commits/4cb1cb5c2411333c370e8d7582c4c13280db4119

**Pull request closed without merging**